bundler/lib/bundler/templates/newgem/circleci/config.yml.tt

```diff
@@ -3,11 +3,21 @@ jobs:
   build:
     docker:
       - image: ruby:<%= RUBY_VERSION %>
+<%- if config[:ext] == 'rust' -%>
+    environment:
+      RB_SYS_FORCE_INSTALL_RUST_TOOLCHAIN: 'true'
+<%- end -%>
     steps:
       - checkout
+<%- if config[:ext] == 'rust' -%>
+      - run:
+          name: Instal Rust/Cargo dependencies
+          command: apt-get update && apt-get install -y clang
+<%- end -%>
       - run:
           name: Run the default task
           command: |
+            gem update --system '<%= ::Gem.rubygems_version %>'
```

Comment:
I'm good with this change, but not sure if it could be controversial for other people since it increases the maintenance burden of the workflow and it doesn't really provide any more determinism (the default RubyGems version is also fixed).

So maybe we should restrict it to Rust gems, which are the ones actually needing this?

Reply:
I don't see any problem with this and there is a little benefit, when using Ruby version like 3.1, you'll get stable rubygems installed for all 3.1.x versions. Anyway feel free to change if that's the only stuff blocking and merge. I'm leaving for today @deivid-rodriguez.

Reply:
The only problem is that now gems generated with `--ext=c` or `--no-ext` (default) have two version numbers that need to update, while before there would be only the Bundler version in there.

Since this PR is focusing on generating CI scripts for `--ext=rust`, and those are the only ones that need an up to date RubyGems, I think it'd be best to keep other kind of gems untouched in this PR.

Reply:
Asked @simi privately whether the above made sense and he said yes so I tweaked the PR myself 👍.